BraveWallet/Settings/CustomNetworkDetailsView.swift

```diff
@@ -0,0 +1,364 @@
+            }
+          }
+        }
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      Section(
+        header:
+          Text(Strings.Wallet.customNetworkBlockExplorerUrlsTitle)
+          .textCase(.none)
+      ) {
+        ForEach(model.blockUrls.indices, id: \.self) { index in
+          NetworkTextField(placeholder: Strings.Wallet.customNetworkUrlsPlaceholder,
+                           item: $model.blockUrls[index]) { newValue in
+            var item: NetworkInputItem = model.blockUrls[index]
+            if validateUrl(newValue) {
+              item.error = nil
+              model.blockUrls[index] = item
+              
+              if index == model.blockUrls.count - 1 { // add a new row
+                let newRow = NetworkInputItem(input: "")
+                model.blockUrls.append(newRow)
+              }
+            } else {
+              item.error = Strings.Wallet.customNetworkInvalidAddressErrMsg
+              model.blockUrls[index] = item
+            }
+          }
+        }
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+    }
+    .navigationBarTitle(Strings.Wallet.customNetworkDetailsTitle)
+    .navigationBarTitleDisplayMode(.inline)
+    .toolbar {
+      ToolbarItemGroup(placement: .confirmationAction) {
+        Button(action: {
+          addCustomNetwork()
+        }) {
+          Text(isEditMode ? Strings.Wallet.update : Strings.Wallet.add)
+            .foregroundColor(Color(.braveOrange))
+        }
+      }
+    }
+  }
+  
+  private func validateUrl(_ url: String) -> Bool {
+    let regex = "http[s]?://(([^/:.[:space:]]+(.[^/:.[:space:]]+)*)|([0-9](.[0-9]{3})))(:[0-9]+)?((/[^?#[:space:]]+)([^#[:space:]]+)?(#.+)?)?"
+    let test = NSPredicate(format: "SELF MATCHES %@", regex)
+    return test.evaluate(with: url)
+  }
+  
+  private func validateAllFields() -> Bool {
+    if model.networkId.input.isEmpty {
+      model.networkId.error = Strings.Wallet.customNetworkEmptyErrMsg
+    }
+    model.networkName.error = model.networkName.input.isEmpty ? Strings.Wallet.customNetworkEmptyErrMsg : nil
+    model.networkSymbolName.error = model.networkSymbolName.input.isEmpty ? Strings.Wallet.customNetworkEmptyErrMsg : nil
+    model.networkSymbol.error = model.networkSymbol.input.isEmpty ? Strings.Wallet.customNetworkEmptyErrMsg : nil
+    if model.networkDecimals.input.isEmpty {
+      model.networkDecimals.error = Strings.Wallet.customNetworkEmptyErrMsg
+    }
+    if var rpcUrl = model.rpcUrls.first {
+      if rpcUrl.input.isEmpty {
+        rpcUrl.error = Strings.Wallet.customNetworkEmptyErrMsg
+        model.rpcUrls[0] = rpcUrl
+      }
+    }
+                 
+    if model.networkId.error != nil
+        || model.networkName.error != nil
+        || model.networkSymbolName.error != nil
+        || model.networkSymbol.error != nil
+        || model.networkDecimals.error != nil
+        || model.rpcUrls.first!.error != nil {
```

Comment:
I assume this is guaranteed but we should probably check somewhere instead of force-unwrapping